File: v3/java/src/main/java/com/konfigthis/newscatcherapi/client/api/SearchApi.java
package com.konfigthis.newscatcherapi.client.api;

import com.konfigthis.newscatcherapi.client.ApiClient;
import com.konfigthis.newscatcherapi.client.ApiException;
import com.konfigthis.newscatcherapi.client.model.*;

import java.util.List;

public class SearchApi extends SearchApiGenerated {

    public SearchApi(ApiClient apiClient) throws IllegalArgumentException {
        super(apiClient);
    }
    public class GetRequestBuilder extends GetRequestBuilderGenerated {
        public GetRequestBuilder(String q) {
            super(q);
        }

        public List<Object> executeSearchAll() {
            int pageSize = this.pageSize != null ? this.pageSize : 100;
            try {
                CSearchResponse response = this.execute();
                if (response.getTotalHits() > pageSize) {
                    for (int page = 2; page <= response.getTotalPages(); page++) {
                        this.page(page);
                        CSearchResponse res = this.execute();
                        response.getArticles().addAll(res.getArticles());
                    }
                }
                return response.getArticles();
            } catch (ApiException e) {
                System.out.printf("Exception when calling Search all: %s\n", e);
                return null;
            }
        }
    }
    public class PostRequestBuilder extends PostRequestBuilderGenerated {
        public PostRequestBuilder(String q) {
            super(q);
        }

        public List<Object> executeSearchAll() {
            int pageSize = this.pageSize != null ? this.pageSize : 100;
            try {
                CSearchResponse1 response = this.execute();
                if (response.getTotalHits() > pageSize) {
                    for (int page = 2; page <= response.getTotalPages(); page++) {
                        this.page(page);
                        CSearchResponse1 res = this.execute();
                        response.getArticles().addAll(res.getArticles());
                    }
                }
                return response.getArticles();
            } catch (ApiException e) {
                System.out.printf("Exception when calling Search all: %s\n", e);
                return null;
            }
        }
    }
}

File: v3/csharp/src/Newscatcherapi.Net/Api/SearchApi.cs
using System;
using System.Reflection;
using System.Collections.Generic;
using Newscatcherapi.Net.Client;
using Newscatcherapi.Net.Model;
using Newscatcherapi.Net.Client;

namespace Newscatcherapi.Net.Api
{
    public partial class SearchApi : SearchApiGenerated {
        public SearchApi() : base() {}
        public SearchApi(string basePath): base(basePath) {}
        public SearchApi(Newscatcherapi.Net.Client.Configuration configuration): base(configuration) {}
        public SearchApi(Newscatcherapi.Net.Client.ISynchronousClient client, Newscatcherapi.Net.Client.IAsynchronousClient asyncClient, Newscatcherapi.Net.Client.IReadableConfiguration configuration): base(client, asyncClient, configuration) {}
    
        public List<Object> GetAll(Dictionary<string, object> parameters) {
            int pageSize = parameters.ContainsKey("pageSize") ? (int) parameters["pageSize"] : 100;

            Type type = this.GetType();
            MethodInfo method = type.GetMethod("Get");
            List<object> paramValues = SearchAllHelper.prepareParameters(parameters, method);

            try {
                SearchGetResponse rawResponse = (SearchGetResponse) method.Invoke(this, paramValues.ToArray());
                DtoResponsesSearchResponseSearchResponse response = rawResponse.GetDtoResponsesSearchResponseSearchResponse();
                if (response.TotalHits > pageSize)
                {
                    for (int page = 2; page <= response.TotalPages; page++)
                    {
                        parameters["page"] = page;
                        SearchGetResponse rawRes = (SearchGetResponse) method.Invoke(this, paramValues.ToArray());
                        DtoResponsesSearchResponseSearchResponse res = rawRes.GetDtoResponsesSearchResponseSearchResponse();
                        response.Articles.AddRange(res.Articles);
                    }
                }
                return response.Articles;
            } catch (Exception e) {
                Console.WriteLine($"Exception when calling Search API: {e}");
                return null;
            }
        }

        public List<Object> PostAll(SearchRequest request) {
            int pageSize = request.PageSize;
            try {
                SearchPostResponse rawResponse = this.Post(request);
                DtoResponsesSearchResponseSearchResponse response = rawResponse.GetDtoResponsesSearchResponseSearchResponse();
                if (response.TotalHits > pageSize)
                {
                    for (int page = 2; page <= response.TotalPages; page++)
                    {
                        request.Page = page;
                        SearchPostResponse rawRes = this.Post(request);
                        DtoResponsesSearchResponseSearchResponse res = rawRes.GetDtoResponsesSearchResponseSearchResponse();
                        response.Articles.AddRange(res.Articles);
                    }
                }
                return response.Articles;
            } catch (Exception e) {
                Console.WriteLine($"Exception when calling Search API: {e}");
                return null;
            }
        }
    }
}